### DISCUSSION TITLE
Spreadsheet to Obsidian Project: CSV to MD Files Script (Python)

### DISCUSSION THREAD

**Discussion by ckep1** [marcusolsson/obsidian-projects#581] [Show and Tell]
I made a Python script to generate the Markdown files used by Projects from a CSV! This creates a file for each row and adds the necessary attributes for each file. 

Usage:
1. Paste the script below into an empty .py file.
2. Save your desired spreadsheet as a CSV file into the same folder as the script.
3. Run the Python script ([basic tutorial](https://geekflare.com/how-to-run-python-scripts/)).
4. Move the resulting `md_files` folder into your Obsidian vault (and rename).
5. In Obsidian, right click the folder and select `Create project in folder`.
6. Success! You should now have a Project to match your input Spreadsheet.

```python
import csv
import os
import re

def sanitize_filename(filename):
    # Remove invalid characters from filename
    return re.sub(r'[\\/*?:"<>|]', '-', filename)

# Directory to store the .md files
output_dir = 'md_files'

# Make sure the output directory exists
if not os.path.exists(output_dir):
    os.makedirs(output_dir)

# Loop through each file in the current directory
for filename in os.listdir('.'):
    # Process only .csv files
    if filename.endswith('.csv'):

        # Read the CSV file
        with open(filename, mode='r', encoding='utf-8') as file:
            reader = csv.DictReader(file)
            headers = reader.fieldnames

            # Iterate through each row in the CSV
            for row in reader:
                # Sanitize and create .md file named after the first column
                title = sanitize_filename(row[headers[0]])
                md_filename = os.path.join(output_dir, f"{title}.md")

                # Open the .md file for writing
                with open(md_filename, 'w', encoding='utf-8') as md_file:
                    # Write attributes to the .md file
                    md_file.write('---\n')
                    for header in headers:
                        if header != headers[0] and row[header]:
                            md_file.write(f"{header}: {row[header]}\n")
                    md_file.write('---\n')

```

**Comment by marcusolsson** [comment]
This is awesome! 😍 Thanks a ton for sharing! 🙏

**Reply by ckep1** [reply]
Thank you for a great plugin, I appreciate your work!

**Comment by rialistic** [comment]
Oh how cool. I guess I'll be able to create projects of full blibiographic CSV-files dowloaded from citation mapping tool (such as WOS, Research Rabbit, Connected paperes). Maybe some tweaks will be needed to have the metadata sit comfortable in the Obsidian-flavoured YAML-system though :)

Thanks a lot!

**Comment by ArthurSrz** [comment]
Thanks a lot @ckep1 ! That's perfect.

**Comment by nwchandler** [comment]
This is super helpful! Thank you for sharing this!
